packages/nightingale-linegraph-track/src/nightingale-linegraph-track.js

```diff
@@ -0,0 +1,149 @@
+import ProtvistaTrack from "protvista-track";
+import * as d3 from "d3";
+import {
+  scaleLinear,
+  select,
+  line,
+  max,
+  min,
+  interpolateRainbow,
+  mouse,
+} from "d3";
+
+class NightingaleLinegraphTrack extends ProtvistaTrack {
+  connectedCallback() {
+    super.connectedCallback();
+
+    this._height = Number(this.getAttribute("height")) || 40;
+    this._yScale = scaleLinear();
+
+    if (this._data) this._createTrack();
+  }
+
+  set data(data) {
+    this._data = data;
+    this._createTrack();
+  }
+
+  _createTrack() {
+    select(this).selectAll("svg").remove();
+    this.svg = select(this)
+      .append("svg")
+      .attr("width", this.width)
+      .attr("height", this._height);
+    this.trackHighlighter.appendHighlightTo(this.svg);
+
+    const range = [];
+    this._data.map((d) => range.push(...d.range));
+    this.minRange = min(range);
+    this.maxRange = max(range);
+
+    // Create the visualisation here
+    this.refresh();
+  }
+
+  refresh() {
```

Comment:
Looking at the old component, I think the use of `refresh` was wrong, as the idea is to use `_createTrack` to add elements (d3's enter) and `refesh` is only supposed to update (anything after the `enter()`). This is so that the elements are only added once, and the only thing that happens is that they are updated when `refresh` is called: currently they are removed and re-added, which causes performance issues. Let me know if it's unclear and we can look at it together.

Reply:
I understand what you mean. I have updated the code in the createTrack to do all the initial creation and assigning data. Refresh only redraws the line with the latest x and y values